packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -139,14 +143,26 @@ export class ValidatorApi implements IValidatorApi {
       return  state.validators.findIndex((v) => this.config.types.BLSPubkey.equals(v.pubkey, publicKey));
     }));
 
-    return validatorIndexes.map((validatorIndex) => {
+    const attesterDuties = validatorIndexes.map((validatorIndex) => {
       return assembleAttesterDuty(
         this.config,
         {publicKey: state.validators[validatorIndex].pubkey, index: validatorIndex},
         state,
         epoch
       );
     });
+    const subnets: Set<string> = new Set();
+    attesterDuties.forEach(duty => subnets.add(getCommitteeIndexSubnet(duty.committeeIndex)));
+    for (const subnet of subnets) {
+      const peerIds = this.reps.getPeerIdsBySubnet(subnet);
+      if (peerIds.length < 3) {
+        // If an insufficient number of current peers are subscribed to the topic,
+        // the validator must discover new peers on this topic
+        this.logger.info(`Found only ${peerIds.length} for subnett ${subnet}, finding new peers to connect`);
+        await this.network.connectToNewPeersBySubnet(parseInt(subnet), peerIds);
+      }
+    }
```

Comment:
Don't do that here, we should start looking only if validator is aggregator which he will notify us by calling "subscribeAttestationSubnet" method.